src/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -69,16 +72,30 @@ public static IPHostEntry GetHostEntry(string hostNameOrAddress)
                 ipHostEntry = GetHostEntryCore(hostNameOrAddress);
             }
 
-            if (NetEventSource.IsEnabled) NetEventSource.Exit(null, ipHostEntry);
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(hostNameOrAddress, $"{ipHostEntry} with {ipHostEntry.AddressList.Length} entries");
             return ipHostEntry;
         }
 
-        public static Task<IPHostEntry> GetHostEntryAsync(string hostNameOrAddress) =>
-            GetHostEntryCoreAsync(hostNameOrAddress, justReturnParsedIp: false, throwOnIIPAny: true);
+        public static Task<IPHostEntry> GetHostEntryAsync(string hostNameOrAddress)
+        {
+            if (NetEventSource.IsEnabled)
+            {
+                NetEventSource.Enter(hostNameOrAddress, hostNameOrAddress);
+                Task<IPHostEntry> t = GetHostEntryCoreAsync(hostNameOrAddress, justReturnParsedIp: false, throwOnIIPAny: true);
+                t.ContinueWith((t, o) => NetEventSource.Exit(hostNameOrAddress, $"{t.Result} with {((IPHostEntry)t.Result).AddressList.Length} entries"),
```

Comment:
If the operation fails, that `t.Result` is going to propagate an exception, and this won't actually log anything.  The resulting task will then also fault and trigger TaskScheduler.UnobservedTaskException.  If your goal is to fire the event on successful exit, you can change the options you pass in to instead be `TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnRanToCompletion`, such that the continuation will only be invoked if the operation completes successfully.

Also, this is allocating an unnecessary closure by referencing `hostNameOrAddress`; it'd be better to pass that in as the object state, e.g.
```C#
t.ContinueWith((t, s) => NetEventSource.Exit((string)s, $"{t.Result} with {((IPHostEntry)t.Result).AddressList.Length} entries"),
    hostNameOrAddress, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnRanToCompletion, TaskScheduler.Default);
```

Reply:
updated. I missed that as I was getting expected trace entries. I verified with AppDomain.CurrentDomain.FirstChanceException that there is no longer exception from task continuation.